dev/ProgressRing/ProgressRingAutomationPeer.cpp

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "ResourceAccessor.h"
+#include "ProgressRing.h"
+#include "ProgressRingAutomationPeer.h"
+#include "Utils.h"
+
+#include "ProgressRingAutomationPeer.properties.cpp"
+
+ProgressRingAutomationPeer::ProgressRingAutomationPeer(winrt::ProgressRing const& owner)
+    : ReferenceTracker(owner)
+{
+}
+
+// IAutomationPeerOverrides
+winrt::IInspectable ProgressRingAutomationPeer::GetPatternCore(winrt::PatternInterface const& patternInterface)
+{
+    if (auto progressRing = Owner().try_as<winrt::ProgressRing>())
+    {
+        if (progressRing.IsIndeterminate())
+        {
+            return nullptr;
+        }
+    }
+
+    if (patternInterface == winrt::PatternInterface::RangeValue)
+    {
+        return *this;
+    }
+
+     return __super::GetPatternCore(patternInterface);
+}
+
+winrt::hstring ProgressRingAutomationPeer::GetClassNameCore()
+{
+    return winrt::hstring_name_of<winrt::ProgressRing>();
+}
+
+winrt::hstring ProgressRingAutomationPeer::GetNameCore()
+{
+    //Check to see if the item has a defined AutomationProperties.Name
+    winrt::hstring name = __super::GetNameCore();
+
+    if (auto progressRing = Owner().try_as<winrt::ProgressRing>())
+    {
+        if (progressRing.IsIndeterminate())
+        {
+            return winrt::hstring{ ResourceAccessor::GetLocalizedStringResource(SR_ProgressRingIndeterminateStatus) + name };
+        }
+    }
+    return name;
+}
+
+winrt::AutomationControlType ProgressRingAutomationPeer::GetAutomationControlTypeCore()
+{
+    return winrt::AutomationControlType::ProgressBar;
```

Comment:
>ProgressBar [](start = 41, length = 11)

Is there a Ring option for this?

Reply:
There isn't a ProgressRing option it seems. In the WUXC version, the AutomationControlType returns ProgressBar, as well.

Reply:
Than I think that's an acceptable solution. From a UIA perspective they are essentially the same. The only real difference is the shape/visuals.

For semantic understanding of the control, users need to know that it indicates progress, and that is what we achieve.